infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationEventListener.java

```diff
@@ -2,26 +2,40 @@
 
 import lombok.extern.slf4j.Slf4j;
 import org.corfudb.infrastructure.logreplication.proto.LogReplicationMetadata.ReplicationEvent;
+import org.corfudb.infrastructure.logreplication.replication.receive.LogReplicationMetadataManager;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.collections.CorfuStore;
 import org.corfudb.runtime.collections.CorfuStreamEntries;
 import org.corfudb.runtime.collections.CorfuStreamEntry;
 import org.corfudb.runtime.collections.StreamListener;
 
+import java.util.Collections;
 import java.util.List;
 
 @Slf4j
 public final class LogReplicationEventListener implements StreamListener {
     private final CorfuReplicationDiscoveryService discoveryService;
+    private final CorfuStore corfuStore;
 
-    public  LogReplicationEventListener(CorfuReplicationDiscoveryService discoveryService) {
+    public  LogReplicationEventListener(
+        CorfuReplicationDiscoveryService discoveryService,
+        CorfuRuntime runtime) {
         this.discoveryService = discoveryService;
+        this.corfuStore = new CorfuStore(runtime);
     }
 
     public void start() {
-        discoveryService.getLogReplicationMetadataManager().subscribeReplicationEventTable(this);
+        log.info("LogReplication start listener for table {}",
+            LogReplicationMetadataManager.REPLICATION_EVENT_TABLE_NAME);
+        corfuStore.subscribeListener(this,
```

Comment:
hmm.. good point.  Since subscription is now triggered from the discovery service, any exception will stop the discovery service.  

I am thinking it may be the right behavior given that exceptions on subscription are only due to incorrect params?  Let us discuss a bit more on this.

Reply:
no conclusion reached yet.  We can discuss this tomorrow.

Reply:
enclosing in a try/catch to prevent interrupting normal replication flow, as discussed offline